pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -545,10 +547,54 @@ private Map<String, InstancePartitions> getTierToInstancePartitionsMap(String ta
    *  InstancePartitions to zk.
    *  Then we'll be able to support replica group assignment while creating InstancePartitions for tiers
    */
-  private InstancePartitions getInstancePartitionsForTier(Tier tier, String tableNameWithType) {
```

Comment:
update the method comment a bit?

also I feel it probably better to keep this method functional, w/o persisting states into ZK as a side effect, which may be put in separate util methods and get called by TableRebalancer based on dryRun/bootstrap flag.

Reply:
Update comments. For the persisting states, I was following this https://github.com/apache/pinot/blob/1ab35ead487a713a126bf03f46604f9d407741c0/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java#L469 as it is for tier side. I'll try to move this out